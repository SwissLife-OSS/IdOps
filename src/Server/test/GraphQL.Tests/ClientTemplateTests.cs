using System.Threading.Tasks;
using HotChocolate;
using HotChocolate.Execution;
using IdOps.Authorization;
using IdOps.Security;
using Snapshooter.Xunit;
using Squadron;
using Xunit;

namespace IdOps.GraphQL.Tests
{
    [Collection(TestCollectionNames.GraphQL)]
    public class ClientTemplateTests : TestHelper
    {
        public ClientTemplateTests(MongoResource resource) : base(resource)
        {
        }

        [Theory]
        [InlineData(Roles.Admin)]
        [InlineData(Roles.Edit)]
        [InlineData(Roles.Read)]
        public async Task ValidRequest(string role)
        {
            // arrange
            ITestRequestBuilder requestBuilder = TestRequestBuilder.New()
                .AddExecutor(await CreateSchema())
                .AddRequestFromFile("GetClientTemplates")
                .AddUser()
                .AddRole(role);

            await TestDataBuilder
                .New(Services)
                .SetupTenant()
                .SetupTemplate()
                .ExecuteAsync();

            // act
            IExecutionResult result = await requestBuilder.ExecuteAsync();

            // assert
            result.ToJson()
                .MatchSnapshot();
        }

        [Fact]
        public async Task GetClientTemplates_NotAuthenticated()
        {
            // arrange
            ITestRequestBuilder requestBuilder = TestRequestBuilder.New()
                .AddExecutor(await CreateSchema())
                .AddRequestFromFile("GetClientTemplates")
                .SetAuthenticated(false);

            await TestDataBuilder
                .New(Services)
                .SetupTenant()
                .SetupTemplate()
                .ExecuteAsync();

            // act
            IExecutionResult result = await requestBuilder.ExecuteAsync();

            // assert
            result.ToJson().MatchSnapshot();
        }
    }
}
