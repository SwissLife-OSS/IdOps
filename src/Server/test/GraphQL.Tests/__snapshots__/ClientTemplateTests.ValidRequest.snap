{
  "data": {
    "clientTemplates": [
      {
        "id": "443680dbca114aa09a5b6eceb1ba1671",
        "name": "Template1",
        "tenant": "TestTenant",
        "apiScopes": [
          "4986e94402f04afeb0768288457b7bf1",
          "d5684d9f86a747f187c161ed2bc3407c"
        ],
        "identityScopes": [
          "2c654c8558a44236bdd5e9ad40beb953"
        ],
        "secrets": [
          {
            "environmentId": "34ca1d86dd6d4837bb9aeeeeb591fef9",
            "type": "SharedSecret",
            "value": "***"
          }
        ]
      }
    ]
  }
}
